Libplanet/Blockchain/BlockLocator.cs

```diff
@@ -1,80 +1,109 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using Libplanet.Blocks;
 
 namespace Libplanet.Blockchain
 {
     /// <summary>
-    /// A class that contains the hashes for a series of blocks.
+    /// A class that contains <see cref="BlockHash"/>es for a series of blocks.
     /// </summary>
     public class BlockLocator : IEnumerable<BlockHash>
     {
-        private readonly IEnumerable<BlockHash> _impl;
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="BlockLocator"/> with a set of indexer
-        /// functions, sampling after <paramref name="sampleAfter"/> number of blocks.
-        /// </summary>
-        /// <param name="indexBlockHash">A function that converts an index to a
-        /// <see cref="BlockHash"/>.</param>
-        /// <param name="indexByBlockHash">A function that converts a <see cref="BlockHash"/> to its
-        /// index.</param>
-        /// <param name="sampleAfter">The number of consequent blocks to include before sampling.
-        /// </param>
-        public BlockLocator(
-            Func<long, BlockHash?> indexBlockHash,
-            Func<BlockHash, long> indexByBlockHash,
-            int sampleAfter = 10
-        )
-            : this(
-                indexBlockHash(-1),
-                indexBlockHash,
-                indexByBlockHash,
-                sampleAfter
-            )
-        {
-        }
+        private readonly List<BlockHash> _impl;
 
         /// <summary>
         /// Initializes a new instance of <see cref="BlockLocator"/> from <paramref name="hashes"/>.
         /// </summary>
         /// <param name="hashes">Enumerable of <see cref="BlockHash"/>es to convert from.</param>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="hashes"/> is empty.
+        /// </exception>
         public BlockLocator(IEnumerable<BlockHash> hashes)
         {
-            _impl = hashes;
+            _impl = hashes.Any()
+                ? hashes.ToList()
+                : throw new ArgumentException($"Given {nameof(hashes)} cannot be empty.");
```

Comment:
```suggestion
                : throw new ArgumentException($"Given {nameof(hashes)} cannot be empty.", nameof(hashes));
```

it may exceed the line limit 😅